app/code/community/Algolia/Algoliasearch/etc/system.xml

```diff
@@ -648,6 +648,20 @@
                         ]]>
                     </comment>
                     <fields>
+                        <enable_synonyms translate="label comment">
+                            <label>Enable Synonyms management</label>
+                            <frontend_type>select</frontend_type>
+                            <source_model>adminhtml/system_config_source_yesno</source_model>
+                            <sort_order>0</sort_order>
+                            <show_in_default>1</show_in_default>
+                            <show_in_website>1</show_in_website>
+                            <show_in_store>1</show_in_store>
+                            <comment>
+                                <![CDATA[
+                                If set to No, synonyms will be managed only in Algolia's dashboard and won't get overriden by Magento
+                                ]]>
+                            </comment>
+                        </enable_synonyms>
                         <synonyms translate="label comment">
                             <label>Synonyms</label>
```

Comment:
I would add a dependency for other synonyms settings, to not display when synonyms is False